src/main/java/lotto/service/LottoServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+package lotto.service;
+
+import camp.nextstep.edu.missionutils.Randoms;
+import lotto.model.Lotto;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/* 실제 LottoService 기능 구현*/
+public class LottoServiceImpl implements LottoService {
+    @Override
+    public List<List<Integer>> generateLottoNumbers(int numberOfLottos) {
+        List<List<Integer>> lottoNumbersList = new ArrayList<>();
+        for (int i = 0; i < numberOfLottos; i++) {
+            List<Integer> lottoNumbers = Randoms.pickUniqueNumbersInRange(1, 45, 6);
+            lottoNumbers.sort(Integer::compareTo);
+            lottoNumbersList.add(lottoNumbers);
+        }
+        return lottoNumbersList;
+    }
+
+    @Override
+    public Map<String, Integer> calculateLotto(List<List<Integer>> lottoNumbers, Lotto winningLotto, int bonusNumber){
+            Map<String, Integer> lottoNumbersMap = new HashMap<>();
+            lottoNumbersMap.put("3개 일치", 0);
+            lottoNumbersMap.put("4개 일치", 0);
+            lottoNumbersMap.put("5개 일치", 0);
+            lottoNumbersMap.put("5개 일치, 보너스 볼 일치", 0);
+            lottoNumbersMap.put("6개 일치", 0);
+
+            for (List<Integer> lottoNumber : lottoNumbers) {
+                int matchCount = getMatchCount(lottoNumber, winningLotto.getNumbers());
+                boolean bonusMatch = lottoNumber.contains(bonusNumber);
+
+                if (matchCount == 6) {
+                    lottoNumbersMap.put("6개 일치", lottoNumbersMap.get("6개 일치") + 1);
+                } else if (matchCount == 5 && bonusMatch) {
+                    lottoNumbersMap.put("5개 일치, 보너스 볼 일치", lottoNumbersMap.get("5개 일치, 보너스 볼 일치") + 1);
+                } else if (matchCount == 5) {
+                    lottoNumbersMap.put("5개 일치", lottoNumbersMap.get("5개 일치") + 1);
+                } else if (matchCount == 4) {
+                    lottoNumbersMap.put("4개 일치", lottoNumbersMap.get("4개 일치") + 1);
+                } else if (matchCount == 3) {
+                    lottoNumbersMap.put("3개 일치", lottoNumbersMap.get("3개 일치") + 1);
+                }
+            }
+        return lottoNumbersMap;
```

Comment:
해당 부분 같은경우에는 이넘 클래스를 이용하여 관리 하는게 효과적이야
```suggestion
    public enum LottoRank {
    FIRST(6, false, 2_000_000_000, "6개 일치"),
    SECOND(5, true, 30_000_000, "5개 일치, 보너스 볼 일치"),
    THIRD(5, false, 1_500_000, "5개 일치"),
    FOURTH(4, false, 50_000, "4개 일치"),
    FIFTH(3, false, 5_000, "3개 일치"),
    NONE(0, false, 0, "");

    private final int matchCount;
    private final boolean matchBonus;
    private final int prize;
    private final String description;
```